test/proxy-call-test.js

```diff
@@ -1176,29 +1176,44 @@ describe("sinonSpy.call", function () {
 
         // https://github.com/sinonjs/sinon/issues/1066
         /* eslint-disable consistent-return */
-        it("does not throw when the call stack is empty", function (done) {
+        it("does not throw when the call stack is empty", async function () {
```

Comment:
This test doesn't have anything to do with the rest of the PR, other than being next to the test I was adding.

I figured since it's just in tests async should be fine to use, but I also just did a search and realized the only other async tests are in `promise-test.js` and an es2015-specific test, so if y'all don't do async tests generally, let me know and I'll revert this bit.

Reply:
We just changed target browsers to evergreen last year, so it's not all that much that has changed since. Adding `async` is fine, as I believe ES2018 is our current ES level.